base/compiler/abstractinterpretation.jl

```diff
@@ -133,6 +153,7 @@ function abstract_call_gf_by_type(interp::AbstractInterpreter, @nospecialize(f),
         this_rt = widenwrappedconditional(this_rt)
         @assert !(this_conditional isa Conditional) "invalid lattice element returned from inter-procedural context"
         seen += 1
+        merged_sig = Union{merged_sig, sig}
```

Comment:
This could be computationally expensive, and probably shouldn't be changed in this PR. Normally we use `fullmatch` here.